src/main/java/org/openstreetmap/atlas/checks/validation/tag/HighwayToFerryTagCheck.java

```diff
@@ -0,0 +1,127 @@
+package org.openstreetmap.atlas.checks.validation.tag;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+import org.openstreetmap.atlas.checks.atlas.predicates.TypePredicates;
+import org.openstreetmap.atlas.checks.base.BaseCheck;
+import org.openstreetmap.atlas.checks.flag.CheckFlag;
+import org.openstreetmap.atlas.geography.atlas.items.AtlasObject;
+import org.openstreetmap.atlas.geography.atlas.items.Edge;
+import org.openstreetmap.atlas.tags.FerryTag;
+import org.openstreetmap.atlas.tags.HighwayTag;
+import org.openstreetmap.atlas.tags.RouteTag;
+import org.openstreetmap.atlas.tags.Taggable;
+import org.openstreetmap.atlas.tags.annotations.validation.Validators;
+import org.openstreetmap.atlas.utilities.configuration.Configuration;
+
+/**
+ * Based on the osm wiki, when route=ferry exists, there should be a ferry=* tag where the values of
+ * ferry are highway values. This check flags all edges that have both {@link HighwayTag} and
+ * {@link RouteTag} value equal to "FERRY".
+ *
+ * @author sayas01
+ */
+public class HighwayToFerryTagCheck extends BaseCheck<Long>
+{
+    private static final String MINIMUM_HIGHWAY_TYPE_DEFAULT = HighwayTag.PATH.toString();
+    private static final String FERRY_TAG_IF_SAME_AS_HIGHWAY_INSTRUCTION = "This way {0,number,#} has a Ferry and a Highway tag for a ferry route. "
+            + "Please verify and remove the highway tag.";
+    private static final String FERRY_TAG_IF_ABSENT_INSTRUCTION = "This way {0,number,#} has a Highway tag for a ferry route instead of a Ferry tag. "
+            + "Please verify and add a Ferry tag with the Highway tag value and remove the highway tag.";
+    private static final String FERRY_TAG_IF_DIFFERENT_FROM_HIGHWAY_INSTRUCTION = "This way {0,number,#} has a Ferry and a Highway tag for a ferry route. "
+            + "Please verify and update the Ferry tag with the Highway tag value and remove the highway tag.";
+    private static final Predicate<Taggable> HAS_FERRY_TAG = object -> Validators
```

Comment:
This will have a problem running on spark because the Predicate interface isn't serializable. You can cast this to ```(Serializeable & Predicate<Taggable>)``` or just create a static function that will return this predicate. I would suggest the static function because its slightly cleaner.

Reply:
I have updated the code by removing the predicate and using Validators directly in the condition check(line [77](https://github.com/osmlab/atlas-checks/blob/96d453987317cd271f93c8d2a9e5fe2026a61ff4/src/main/java/org/openstreetmap/atlas/checks/validation/tag/HighwayToFerryTagCheck.java#L77)).